AdminUi/src/AdminUi/ClientApp/src/app/components/quotas/identity/identity-edit/identity-edit.component.ts

```diff
@@ -49,18 +53,50 @@ export class IdentityEditComponent {
             next: (data: HttpResponseEnvelope<Identity>) => {
                 if (data && data.result) {
                     this.identity = data.result;
+                    this.groupQuotasByMetricForTable();
                 }
             },
             complete: () => (this.loading = false),
             error: (err: any) => {
                 this.loading = false;
-                this.snackBar.open(err.message, "Dismiss", {
-                    panelClass: ["snack-bar"]
+                let errorMessage = err.error?.error?.message ?? err.message;
+                this.snackBar.open(errorMessage, "Dismiss", {
+                    verticalPosition: "top",
+                    horizontalPosition: "center"
                 });
             }
         });
     }
 
+    groupQuotasByMetricForTable() {
+        let quotas = [...this.identity.quotas];
+        this.quotasTableData = [];
+
+        quotas.sort((a, b) => a.metric.key.localeCompare(b.metric.key) || a.source.localeCompare(b.source));
+        while (quotas.length > 0) {
+            this.quotasTableData.push({
+                metric: quotas[0].metric,
+                isGroup: true
+            } as MetricGroup);
+            quotas = this.iterateQuotasByMetric(quotas, quotas[0].metric.key);
+        }
+    }
+
+    iterateQuotasByMetric(quotas: Quota[], metricKey: string): Quota[] {
+        if (quotas.length == 0) return [];
+
+        if (quotas[0].metric.key == metricKey) {
+            this.quotasTableData.push(quotas[0]);
+            return this.iterateQuotasByMetric(quotas.slice(1), metricKey);
+        }
+
+        return quotas;
+    }
+
+    isGroup(index: any, item: any): boolean {
```

Comment:
This method isn't used anywhere. Is it used by angular?

Reply:
It's used for determining whether the table row is a quota or a metric group:
`<tr mat-row *matRowDef="let row; columns: ['metricGroup']; when: isGroup"></tr>`